lib/modules/service-media.js

```diff
@@ -222,6 +222,65 @@ OnvifServiceMedia.prototype.getVideoEncoderConfiguration = function(params, call
 	}
 };
 
+/* ------------------------------------------------------------------
+* Method: setVideoEncoderConfiguration(configuration, forcePersistence[callback])
+* - params:
+*   - configuration | Object | required | all camera configs
+*   - forcePersistence | Boolean | optional | deprecated - always considered to be true
+* ---------------------------------------------------------------- */
+OnvifServiceMedia.prototype.setVideoEncoderConfiguration = function(configuration, forcePersistence, callback) {
+	let promise = new Promise((resolve, reject) => {
+		let soap_body = '';
+		soap_body += `<trt:SetVideoEncoderConfiguration>
+						<trt:Configuration token="${configuration.$.token}">
+							<tt:Name>${configuration.Name}</tt:Name>
+							<tt:UseCount>${configuration.UseCount}</tt:UseCount>
+							<tt:Encoding>${configuration.Encoding}</tt:Encoding>
+							<tt:Resolution>
+								<tt:Width>${configuration.Resolution.Width}</tt:Width>
+								<tt:Height>${configuration.Resolution.Height}</tt:Height>
+							</tt:Resolution>
+							<tt:Quality>${configuration.Quality}</tt:Quality>
+							<tt:RateControl>
+								<tt:FrameRateLimit>${configuration.RateControl.FrameRateLimit}</tt:FrameRateLimit>
+								<tt:EncodingInterval>${configuration.RateControl.EncodingInterval}</tt:EncodingInterval>
+								<tt:BitrateLimit>${configuration.RateControl.BitrateLimit}</tt:BitrateLimit>
+							</tt:RateControl>
+							<tt:H264>
+								<tt:GovLength>${configuration.H264.GovLength}</tt:GovLength>
+								<tt:H264Profile>${configuration.H264.H264Profile}</tt:H264Profile>
+							</tt:H264>
+							<tt:Multicast>
+								<tt:Address>
+									<tt:Type>${configuration.Multicast.Address.Type}</tt:Type>
+									<tt:IPv4Address>${configuration.Multicast.Address.IPv4Address}</tt:IPv4Address>
+								</tt:Address>
+								<tt:AutoStart>${configuration.Multicast.AutoStart}</tt:AutoStart>
+								<tt:Port>${configuration.Multicast.Port}</tt:Port>
+								<tt:TTL>${configuration.Multicast.TTL}</tt:TTL>
+							</tt:Multicast>
+							<tt:SessionTimeout>${configuration.SessionTimeout}</tt:SessionTimeout>
+						</trt:Configuration>
+			        	<trt:ForcePersistence>${forcePersistence ? forcePersistence : true}</trt:ForcePersistence>
```

Comment:
if I pass "false" to falsePersistence – here we will have true anyway. Seems like 
`forcePersistence !== undefined ? forcePersistence : true` would be better here.

Reply:
I think you got it right, was a mistake of my part